test/puppetlabs/trapperkeeper/testutils/logging_test.clj

```diff
@@ -0,0 +1,150 @@
+      (is (= (set (cons new-appender original-appenders))
+             (set (get-appenders logger))))
+      (log/error uuid))
+    (is (= (set original-appenders)
+           (set (get-appenders logger))))
+    (is (some #(= expected %) (map event->map @log)))))
+
+(deftest with-log-appenders
+  (let [log (atom [])
+        logger (find-logger root-logger-name)
+        uuid (kitchensink/uuid)
+        original-appenders (get-appenders logger)
+        new-appender (log-event-listener (fn [event] (swap! log conj event)))
+        expected {:logger "puppetlabs.trapperkeeper.testutils.logging-test"
+                  :level :error
+                  :message uuid
+                  :exception nil}]
+    (tgt/with-log-appenders root-logger-name
+      [new-appender]
+      (is (= [new-appender] (get-appenders logger)))
+      (log/error uuid))
+    (is (= (set original-appenders)
+           (set (get-appenders logger))))
+    (is (some #(= expected %) (map event->map @log)))))
+
+(deftest with-log-listener
+  (let [log (atom [])
+        uuid (kitchensink/uuid)
+        expected {:logger "puppetlabs.trapperkeeper.testutils.logging-test"
+                  :level :info
+                  :message uuid
+                  :exception nil}]
+    (tgt/with-log-level root-logger-name :info
+      (tgt/with-log-event-listener root-logger-name
+        (fn [event] (swap! log conj event))
+        (log/info uuid))
+      (is (some #(= expected %) (map event->map @log))))))
+
+(deftest suppressing-log-unless-error
+  (let [uuid (kitchensink/uuid)
+        target (format "some random message %s" uuid)]
+    (testing "log not dumped if uninteresting"
+      (is (not (re-find (re-pattern target)
+                        (with-out-str
+                          (binding [*err* *out*]
+                            (tgt/with-log-suppressed-unless-notable
+                              (constantly false)
+                              (log/info target))))))))
+    (testing "log dumped if notable"
+      (is (re-find (re-pattern target)
+                   (with-out-str
+                     (binding [*err* *out*]
+                       (tgt/with-log-suppressed-unless-notable
+                         #(= "lp0 on fire" (:message (event->map %)))
+                         (log/info target)
+                         (log/info "lp0 on fire")))))))))
+
+(deftest with-test-logging
+  (testing "basic matching"
+    (doseq [[item test] [["foo" "foo"
+                          "barbar" #"rb"
+                          "baz" (fn [e]
+                                  (and (= :trace (:level e))
+                                       (= "baz" (:message e))))]]]
+      (tgt/with-test-logging
+        (log/trace item)
+        (is (logged? test)))
+      (tgt/with-test-logging
+        (log/trace "hapax legomenon")
+        (is (not (tgt/logged? test))))))
+  (testing "level matches"
+    (doseq [level @#'puppetlabs.trapperkeeper.testutils.logging/levels]
```

Comment:
I believe because it's a private var (levels and level-keywords are just internal "caches").  And removing it produces "java.lang.IllegalArgumentException: Don't know how to create ISeq from: clojure.lang.Var".